### PR TITLE
persistent-temp-files-plugin: (persistent-untitled-documents) added p…

### PR DESCRIPTION
**Pull request by LiquidCake**
…lugin description

### CODE REVIEW

File: doc/geany.txt [hunk 1]
```diff
@@ -5371,22 +5371,42 @@ a given delay.
...
 Save on focus out
-^^^^^^^^^^^^^^^^^
+`````````````````
 
 You can save the current document when the editor's focus goes out.
 Every pop-up, menu dialogs, or anything else that can make the editor lose the focus,
 will make the current document to be saved.
 
 
+Untitled Document Save
```
[the remark this answers is not in the document]
Comment (LiquidCake) [reply doc/geany.txt]: done

File: doc/geany.txt [hunk 2]
```diff
@@ -5371,22 +5371,42 @@ a given delay.
...
-This plugin sets on every new file (*File->New* or *File->New (with template)*)
-a randomly chosen filename and set its filetype appropriate to the used template
-or when no template was used, to a configurable default filetype.
-This enables you to quickly compile, build and/or run the new file without the
-need to give it an explicit filename using the Save As dialog. This might be
-useful when you often create new files just for testing some code or something
-similar.
+````````````
+
+In this mode, for each new editor tab opened by the user - plugin creates a new
```
[the remark this answers is not in the document]
Comment (LiquidCake) [reply doc/geany.txt]: done

File: doc/geany.txt [hunk 3]
```diff
@@ -5371,22 +5371,42 @@ a given delay.
...
+file with randomly generated name inside a configured directory.
```
[the remark this answers is not in the document]
Comment (LiquidCake) [reply doc/geany.txt]: done

File: doc/geany.txt [hunk 4]
```diff
@@ -5371,22 +5371,42 @@ a given delay.
...
+This enables user to quickly compile, build and/or run the new file without the
+need to assign it an explicit filename through the Save As dialog.
```
[the remark this answers is not in the document]
Comment (LiquidCake) [reply doc/geany.txt]: done

File: doc/geany.txt [hunk 5]
```diff
@@ -5371,22 +5371,42 @@ a given delay.
...
-or when no template was used, to a configurable default filetype.
-This enables you to quickly compile, build and/or run the new file without the
-need to give it an explicit filename using the Save As dialog. This might be
-useful when you often create new files just for testing some code or something
-similar.
+````````````
+
+In this mode, for each new editor tab opened by the user - plugin creates a new
+file with randomly generated name inside a configured directory.
+This enables user to quickly compile, build and/or run the new file without the
```
[the remark this answers is not in the document]
Comment (LiquidCake) [reply doc/geany.txt]: done

File: doc/geany.txt [hunk 6]
```diff
@@ -5371,22 +5371,42 @@ a given delay.
...
+need to assign it an explicit filename through the Save As dialog.
+
+By default, OS temp directory is used so files are cleaned up after reboot.
```
[the remark this answers is not in the document]
Comment (LiquidCake) [reply doc/geany.txt]: done

File: doc/geany.txt [hunk 7]
```diff
@@ -5371,22 +5371,42 @@ a given delay.
...
+
+Persistent Untitled Documents
+`````````````````````````````
+
+Goal of this mode is to automatically persist the contents of newly-created editor
+tab, so user could just spawn it, add some text and close Geany in 1 click without
+having to explicitly save current work through the Save As dialog.
+Tab will re-appear on next Geany launch with all its contents restored.
+
+Geany manages and cleans up an underlying storage files automatically.
```
[the remark this answers is not in the document]
Comment (LiquidCake) [reply doc/geany.txt]: done

File: doc/geany.txt [hunk 8]
```diff
@@ -5371,22 +5371,42 @@ a given delay.
...
 
 You can save the current document when the editor's focus goes out.
 Every pop-up, menu dialogs, or anything else that can make the editor lose the focus,
 will make the current document to be saved.
 
 
+Untitled Document Save
+^^^^^^^^^^^^^^^^^^^^^^
+
+This plugin automatically creates an underlying file for newly-opened editor tabs
```
[the remark this answers is not in the document]
Comment (LiquidCake) [reply doc/geany.txt]: done

**Comment by LiquidCake** [comment]
applied all proposed fixes

**Comment by techee** [comment]
Looks good, thanks. With the documentation of templates, let's wait how what I proposed in https://github.com/geany/geany/pull/3911#issuecomment-2504626664 turns out.

**Review by techee — APPROVED** [review]

**Pull request merged**